Automattic/wp-calypso
# Social Preview: Update Facebook Preview

Related to 1203820933396971-as-1204281808721008

## Proposed Changes

This PR updates the `FacebookPreview` component of the `social-previews` package to match the existing Facebook site and post previews. It also updates the components instantiating `FacebookPreview` accordingly. Further changes will be made in the [jetpack monorepo](https://github.com/automattic/jetpack) to update the preview in WPadmin.

## Implementation Notes

- The render of the various previews was determined based on the actual renders on Facebook (see captures below).
- Facebook seems to truncate texts so that they don't take a certain amount of lines (or so that some elements don't exceed a certain height). This implementation uses a maximum number of characters instead. It's a good approximation for now, but that could be improved in a subsequent PR.
- Links in the custom message are not supported yet. It will be handled in a separate PR.
- Tests will be completed in a later PR.

## Testing Instructions

### Prerequisites
- Make sure you have a Facebook account.
- Make sure you have a self-hosted site connected to Jetpack, with a subscription that includes Social.
- From the _Social_ or _My Jetpack_ page of your site WPadmin, activate Social.
- Then, connect your site to Facebook (check [these instructions](https://jetpack.com/support/jetpack-social/connecting-to-social-networks/) if you need help). You may need to create a test page.
- Spin up WordPress.com, by using a live link below or by running this branch locally.

### Testing a site

- Visit `marketing/traffic/:site`
- In the _Website Meta_ section, click _Show Previews_
- Select the Facebook Preview
- The preview of your site should look like this:
<img width="500" alt="Screenshot 2023-04-11 at 10 04 09 AM" src="https://user-images.githubusercontent.com/1620183/231282469-9d487db8-3716-4746-bcd7-29880ef39ad9.png">

### Testing a post

- Create a new post from your site WPadmin
- In the Jetpack sidebar, make sure the post will be published automatically to your Facebook page
<img width="395" alt="Screenshot 2023-04-11 at 4 43 17 PM" src="https://user-images.githubusercontent.com/1620183/231283940-732fc281-184d-40f5-9415-22ef3c7c8e53.png">

- Publish it
- Visit `/posts/:site` in Calypso
- You should see the newly created post under the _Published_ tab
- In the post menu, click _Share_, then press the _Preview_ button
<img width="500" alt="Screenshot 2023-04-11 at 4 47 11 PM" src="https://user-images.githubusercontent.com/1620183/231284467-7d3c5008-321b-444f-a3f9-8b732eaa323c.png">

- Select the Facebook preview
- Notice that it looks like the preview on Facebook, with the caveats mentioned in the **Implementation Notes** section
- Optionally, test different post variations as illustrated below

_Don't mind the fact that the user actions are not shown in the Facebook.com captures._
| Post | WordPress.com | Facebook.com |
|------|-----------------|----------------|
| Title only |<img width="300" alt="Screenshot 2023-04-11 at 10 08 12 AM" src="https://user-images.githubusercontent.com/1620183/231285194-ec2d01d4-19b3-463f-812c-320ed50c8f9a.png">|<img width="300" alt="Screenshot 2023-04-11 at 10 08 55 AM" src="https://user-images.githubusercontent.com/1620183/231285196-fe667099-d4bd-45b0-ac1e-b3cdbb26a168.png">|
| Description only |<img width="300" alt="Screenshot 2023-04-11 at 10 10 19 AM" src="https://user-images.githubusercontent.com/1620183/231285604-90590f29-1e3e-485e-bf6f-5f68ab353c0c.png">|<img width="300" alt="Screenshot 2023-04-11 at 10 10 26 AM" src="https://user-images.githubusercontent.com/1620183/231285606-c065c90c-34fb-44d7-9ca5-1f9ee0388dbb.png">|
| Title & Description |<img width="300" alt="Screenshot 2023-04-11 at 10 12 31 AM" src="https://user-images.githubusercontent.com/1620183/231285798-fbb0d815-3d41-45d5-bfdf-410c4bddd67f.png">|<img width="300" alt="Screenshot 2023-04-11 at 10 11 38 AM" src="https://user-images.githubusercontent.com/1620183/231285796-44d8129b-2370-49b6-93e7-086edc6d77bb.png">|
| Excerpt |<img width="300" alt="Screenshot 2023-04-11 at 10 21 42 AM" src="https://user-images.githubusercontent.com/1620183/231286002-9b6fc16c-3f37-4280-99d0-d1cf36aa4124.png">|<img width="300" alt="Screenshot 2023-04-11 at 10 14 40 AM" src="https://user-images.githubusercontent.com/1620183/231286001-218bc53a-795e-4466-9e99-9ae5485e0fab.png">|
| Custom message |<img width="300" alt="Screenshot 2023-04-11 at 11 00 50 AM" src="https://user-images.githubusercontent.com/1620183/231286105-9b71a428-47e5-4505-ae85-25935f76ca48.png">|<img width="300" alt="Screenshot 2023-04-11 at 11 00 38 AM" src="https://user-images.githubusercontent.com/1620183/231286104-56c61055-5a05-4d65-a344-9f8452e755d9.png">|
| Landscape image |<img width="300" alt="Screenshot 2023-04-11 at 3 37 49 PM" src="https://user-images.githubusercontent.com/1620183/231286477-f939999c-0a13-4e54-8dca-020079098bc3.png">|<img width="300" alt="Screenshot 2023-04-11 at 3 37 19 PM" src="https://user-images.githubusercontent.com/1620183/231286468-0784d049-2641-461a-90a7-6d13c2f23127.png">|
| Portrait Image |<img width="300" alt="Screenshot 2023-04-11 at 3 47 47 PM" src="https://user-images.githubusercontent.com/1620183/231286526-4bf76900-9236-49cd-b3c8-43355559cf08.png">|<img width="300" alt="Screenshot 2023-04-11 at 3 42 25 PM" src="https://user-images.githubusercontent.com/1620183/231286524-fa6fa842-46df-4d2c-b579-93148f30e476.png">|

## Pre-merge Checklist




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204281808721008
